### PR TITLE
Bugfix/django tests pass locally

### PR DESCRIPTION
**Pull request by gecBurton**
## Context

As an engineer I want the django tests to work locally so that I can debug the steps involved

## Things to check

- [x] I have added any new ENV vars in all deployed environments
- [x] I have tested any code added or changed
- [ ] I have run integration tests

### CODE REVIEW

File: django_app/redbox_app/redbox_core/client.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def s3_client():
             aws_secret_access_key=settings.AWS_S3_SECRET_ACCESS_KEY,
             region_name=settings.AWS_S3_REGION_NAME,
         )
-    elif settings.OBJECT_STORE == "minio":
+    else:
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/client.py:16]: we always want a client. Django cant/doesnt use moto (at the moment)

File: .env.test [hunk 2]
```diff
@@ -58,7 +58,7 @@ POSTGRES_DB=redbox-core
 POSTGRES_PASSWORD=insecure
 CONTACT_EMAIL=redbox-copilot@cabinetoffice.gov.uk
 POSTGRES_HOST=localhost
-CORE_API_HOST=http://localhost
+CORE_API_HOST=testserver
```
Comment (gecBurton) [inline .env.test:61]: this is the trigger to mock the core-api

File: django_app/tests/test_views.py [hunk 3]
```diff
@@ -26,6 +26,8 @@ def count_s3_objects(s3_client) -> int:
 
 @pytest.mark.django_db
 def test_upload_view(alice, client, file_pdf_path, s3_client):
+    s3_client.delete_object(Bucket=settings.BUCKET_NAME, Key=file_pdf_path.split("/")[-1].replace(" ", "_"))
```
Comment (gecBurton) [inline django_app/tests/test_views.py:29]: if the file already exist then the file-count wont increase as the file will just be overwritten

File: .env.test (hunk 2 above)
Comment (brunns) [reply .env.test:61]: This will break using `.env.test` to run the app locally, right?

Comment (brunns) [reply .env.test:61]: I've just checked - it does.

File: django_app/redbox_app/redbox_core/client.py [hunk 4]
```diff
@@ -39,26 +39,23 @@ def __init__(self, host: str, port: int):
...
         response = requests.post(
             f"{self.url}/file", json={"key": name}, headers={"Authorization": user.get_bearer_token()}, timeout=30
         )
         if response.status_code != 201:
             raise ValueError(response.text)
         return response.json()
 
-    def rag_chat(self, message_history: list[dict[str, str]], token: str) -> str:
-        url = f"{self.url}/chat/rag"
+    def rag_chat(self, message_history: list[dict[str, str]], user: User) -> str:
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/client.py:52]: changed this for consistency with `embed_file`

**Comment by gecBurton** [comment]
> Works great - I'm just concerned about local running.

good point, this is now addressed via mocking

File: django_app/redbox_app/redbox_core/views.py [hunk 5]
```diff
@@ -111,14 +111,13 @@ def upload_view(request):
                 errors.append(value_error.args[0])
 
             try:
-                api.upload_file(settings.BUCKET_NAME, uploaded_file.name, request.user)
+                api.embed_file(uploaded_file.name, request.user)
```
Comment (gecBurton) [inline django_app/redbox_app/redbox_core/views.py:114]: a more meaningful name?

File: django_app/redbox_app/redbox_core/client.py [hunk 6]
```diff
@@ -39,26 +39,23 @@ def __init__(self, host: str, port: int):
...
         response = requests.post(
-            url, json={"message_history": message_history}, headers={"Authorization": token}, timeout=60
+            f"{self.url}/chat/rag",
+            json={"message_history": message_history},
+            headers={"Authorization": user.get_bearer_token()},
+            timeout=60,
         )
+        if response.status_code != 200:
```
Comment (brunns) [inline django_app/redbox_app/redbox_core/client.py:59]: Consider `request.raise_for_status()`?

**Review by brunns — APPROVED** [review]
Only the tiniest of nitpicks - feel free to ignore.

**Pull request merged**